src/html5-qrcode.ts

```diff
@@ -502,6 +511,63 @@ export class Html5Qrcode {
         });
     }
 
+    /**
+     * Checks if current video track has flash light capability.
+     * 
+     * This feature requires that camera is started.
+     *
+     * @returns Promise with detected flash capability result.
+     */
+    public hasFlash(): Promise<boolean> {
+        if (!("ImageCapture" in window)) {
+            return Promise.resolve(false);
+        }
+
+        const track = this.videoElement?.srcObject ? (<MediaStream>this.videoElement.srcObject).getVideoTracks()[0] : null;
```

Comment:
nit, split into two lines?

Reply:
Comment 2: could this lead to null ptr exception if `...getVideoTracks()` returns undefined or empty array? we should do a `length` check first.